cpp/include/cudf/wrappers/timestamps.hpp

```diff
@@ -36,19 +36,22 @@ using time_point = simt::std::chrono::time_point<simt::std::chrono::system_clock
 
 template <class Duration>
 struct timestamp : time_point<Duration> {
+  // Bring over base class constructors and make them visible here
+  using time_point<Duration>::time_point;
+
+  // This is needed as __shared__ objects of this type can't be assigned in device code
+  // when the initializer list constructs subobjects with values, which is what std::time_point
+  // does.
   constexpr timestamp() : time_point<Duration>(Duration()){};
-  constexpr timestamp(Duration d) : time_point<Duration>(d){};
+
+  // Implicitly convert a tick count into a timestamp
+  // TODO: is this still needed and is this operation even meaningful? The duration units
+  // cannot be inferred from this
```

Comment:
yes, duration is known via `T::duration`, where T is the timestamp. as you said, disallowing this may require a large number of changes, mainly in the area of typed tests that uses the entire range of fixed width types. perhaps, this can be a follow-up pr.

Reply:
Since we're already here modifying the timestamp constructors list, I vote we remove the additional ones now, with the exception of the one necessary to support `__shared__` values.

Reply:
i'm for removing `constexpr timestamp(typename Duration::rep r)`. however, can we do this as a follow up pr just because of the sheer number of changes this will bring about?

Reply:
Can we remove this constructor now that the `fixed_width_column_wrapper` test utility has the `SourceElementT` type parameter? Or is it still needed for other reasons?

Reply:
there are more changes required if that constructor is removed, which i'm working on as a separate pr.